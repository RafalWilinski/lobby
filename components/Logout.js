import React from "react";
import { BarLoader } from "react-spinners";
import HVCenterWrapper from "./HVCenterWrapper";

class Logout extends React.PureComponent {
  componentDidMount() {
    if (window.localStorage) {
      localStorage.removeItem("user");
      this.props.logout();
    }
  }

  render() {
    if (!this.props.isLoggedIn && this.props.url) {
      this.props.url.push({
        pathname: "/login"
      });
    }

    return (
      <HVCenterWrapper>
        <BarLoader loading color={"#108ee9"} />
      </HVCenterWrapper>
    );
  }
}

export default Logout;
