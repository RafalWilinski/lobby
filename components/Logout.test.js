import React from "react";
import renderer from "react-test-renderer";
import Logout from "./Logout";

describe("Logout", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Logout />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
