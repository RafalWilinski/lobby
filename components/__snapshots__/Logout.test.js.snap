// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout renders correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "100vh",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <div
    className="css-1mau7k7"
  >
    <div
      className="css-1di9y0u"
    />
    <div
      className="css-10959i8"
    />
  </div>
</div>
`;
